### DISCUSSION TITLE
Avoid scrolling in option menus

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#496] [Q&A] [answered]
My smartphone hit the ground and crashed - NO, not while riding a motorcycle! Now I take for few days a spare phone ...
When using it, I noticed that the route option menu became scrollable:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/d59d81b2-3f61-4c84-b9e5-9776111beb3f"/>
not easy to use, because it suddenly disappears when you move it forward. 

Question: couldn't all option menus be automatically reduced in size so that they always fit completely on the screen? Even if this means that the buttons are a bit smaller?

**Comment by Ciutnik** [comment]
I have a slightly different problem, also related to scrolling through the menu. You can't scroll the window  `Application | Display |  Arrangement of panels` at all because sliding your finger changes the position of the list items.
This always happens in landscape mode, perhaps in some cases in portrait mode as well. @WolfgangZ42, check if you can see the last item (I guess he should be `GPS recording`).

**Reply by WolfgangZ42** [reply]
Yes - same here. In portrait mode the last line is "Odometer" (I'll never use it at all). But I can't adjust "GPS recording" because it's no more visible.  Maybe we could show only selected panels in the arrangement list?

**Reply by Ciutnik** [reply]
> Maybe we could show only selected panels in the arrangement list?

Next setting?... Alternatively, return (?) to moving items in lists by touching only their icons (I think it was like this before). It seems to me that rearranging the menu is not something that is often done and needs to be easily accessible.

**Reply by WolfgangZ42** [reply]
No - no next settings ...
But all items in this list with status "Off" 
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/26baf5bc-3cbb-44aa-886a-1d703ff7678c"/>
don't need any arrangement - so the arrangement list could be short, except there is an user who will see everything. 🥴

**Reply by Ciutnik** [reply]
> But all items in this list with status "Off" don't need any arrangement

Very good point! The problem may be that each element can be turned on or off independently for three different operating modes: map viewing, navigation, following. This complicates things a bit 🙂

**Reply by devemux86** [reply]
> each element can be turned on or off independently for three different operating modes: map viewing, navigation, following. This complicates things a bit 🙂

Indeed, this is not so simple. :slightly_smiling_face:

**Comment by devemux86** [comment]
> You can't scroll the window Application | Display |  Arrangement of panels at all because sliding your finger changes the position of the list items.

Thanks for the report, this is easy and will be fixed in the next update.
Usually only the icon should reorder and the rest row should scroll list.

**Comment by devemux86** [comment]
> Question: couldn't all option menus be automatically reduced in size so that they always fit completely on the screen? Even if this means that the buttons are a bit smaller?

I will check what is possible. Also the routing menu now has 10 items.
Another alternative is to not go above 8 items per menu screen.

**Comment by devemux86** [comment]
@WolfgangZ42 have you changed the display or font size in application settings?

We can try with smaller icons so more items can fit vertically in the screen without scrolling,
but increasing the app size settings will always cause such issues.

**Reply by WolfgangZ42** [reply]
I set the panel size to 120 % - now to 100 %. Both show the same scolling behaviour.

**Reply by devemux86** [reply]
@WolfgangZ42 what is the device model?

**Reply by WolfgangZ42** [reply]
It's a Sony Xperia XZ2 (H8216) 
I think "panel size" is only relevant for the navigation panels (left/right).

**Reply by devemux86** [reply]
> Sony Xperia XZ2

5.7 inches, 1080 x 2160 pixels, this screen should display everything.

It is the display and font size in _application advanced settings_.
Have you changed any of them?

**Reply by WolfgangZ42** [reply]
No - never changed that.

**Reply by devemux86** [reply]
@WolfgangZ42 do you have them "System" or "100%"?

If they are "System", have you increased the Display or Font size in _Android display settings_?

**Reply by WolfgangZ42** [reply]
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/2fe6d7b0-b847-4259-8a04-42499d22297b"/>
<img height="200" src="https://github.com/devemux86/cruiser/assets/121806238/cfb0d2d4-b2b9-4047-b50f-e2b6b5a87b12"/>
As you see it's slightly in scrolling mode - even more with the navi smartphone of my friend a Sony Xperia XZ.

**Reply by WolfgangZ42** [reply]
It's set to "100 %"

**Reply by devemux86** [reply]
In such cases we can also remove the title, it will give us several pixels available, especially in landscape.

**Comment by devemux86** [comment]
For 7+ items we can also have 3 items per row:

<img src="https://github.com/devemux86/cruiser/assets/3484020/3a62e4cb-7f4f-43fc-b35f-dad69d987900" width="50%"/>

**Reply by devemux86** [reply]
This will also help in navigation when we select waypoint for the info panel or rerouting,
when the route has too many via and shaping points.

(especially with the new function of [auto filled shaping points](https://github.com/devemux86/cruiser/discussions/499))

**Reply by DIrgi** [reply]
When there are 7+ items, maybe text size should be one or two points lower, otherwise we have probably in some languages a few unsightly text breaks - see picture:

![image](https://github.com/devemux86/cruiser/assets/57968526/a6bb3088-3d48-4de5-9643-6af096a23d40)
![image](https://github.com/devemux86/cruiser/assets/57968526/8c47d430-0061-4fe7-b41f-18a629af4c63)
![image](https://github.com/devemux86/cruiser/assets/57968526/d085f667-517a-4bf0-9c13-3aa146f8e0a7)

or like here:
![image](https://github.com/devemux86/cruiser/assets/57968526/626e66ee-c585-4176-b994-744c51e43e60)
better using a blank instead of a "-" like in "Shaping-Point 3" -> better "Shaping Point 3" to have a clear system line break

If proposal is not possible, maybe we should revise the words. i.e. in German (although it's right) you need not to write "Routendiagramm", you can write "Routen Diagramm" as well to have a better break. Or some words can be even shorter like:
"Nahegelegene Favoriten" --> "Nahegel. Favoriten" or "Nahe Favoriten"......

**Reply by bios4** [reply]
> in German (although it's right) you need not to write "Routendiagramm", you can write "Routen Diagramm" as well to have a better break. Or some words can be even shorter like:
"Nahegelegene Favoriten" --> "Nahegel. Favoriten" or "Nahe Favoriten"......

While I agree that the (shown above) text breaks for German translations are ugly, I STRONGLY DISAGREE with the proposal to "divide" good-sounding translations to weird ones: "Routen Diagramm" for example is for sure not a suitable translation, so please DON'T do this...

Shorten some entries like the example with favorites is an option, tough.

BTW @devemux86: how does the app decide whether to show three menu item rows or not?

It always shows the default "two tile menus" on my devices...
With my font settings I also don't get the strange text breaks.

Maybe you @DIrgi could collect all affected translations and we could think about shortening them?

Or - @devemux86 - is there a setting where you can "force" a line break at a certain position of a translation?
Then we could go through the translations and add this forced line breaks at the proper position...

**Reply by DIrgi** [reply]
You see this when you install the latest Beta Version which is B10 where there are more then 6 icons in some masks as shown.

Shortening make only sense when everyone will understand it. And as I said this issue will probably pop up in all languages more or less except English since Emux probably check this while programming.
I would skip in some cases language rules when it’s in the end better readable and understandable. But I agree that there should be no automatism- only where it’s necessary due to smaller icon space. A break from „Routendiagramm“ into „Routen Diagramm“ (only for this icons) is in my opinion more clearer then „RTdiagr.“ or „RT Diagramm“ or just „Diagramm“

**Reply by bios4** [reply]
Ah OK, I'm still at B9...

> „RTdiagr.“ or „RT Diagramm“ or just „Diagramm“

I agree, such ^^ mutilations make no sense at all.

Nevertheless my preferred solution in SUCH a case would be a proper line break "Routen-diagramm" (with "-diagramm" shown in the next line), instead of these ugly word divisions.

Therefore I asked for the possibility for a forced line break, let's see what Emux answers.

**Reply by bios4** [reply]
> Ah OK, I'm still at B9...

Updated to B10 now - the line breaks for the German translations in the "three tile menu" for sure are a step backwards compared to the line breaks in the "two tile menu" from B9:

<img src="https://github.com/devemux86/cruiser/assets/145460147/b0e991b1-4e62-4f42-8322-12ab494d4645" width="49%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/4c4075dd-af6c-4c4a-9627-c86e093989b3" width="49%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/f4d82e87-fd97-4fe8-9ee6-a76df8040357" width="49%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/ee88d726-5d63-421f-9ba0-b966a7081f4e" width="49%">

I personally like the two-tile menu way more anyway...

The three tiles in a row are too dense from my POV and make the menu less clearly arranged compared to the old two-tile approach.

Could this be somehow a setting @devemux86?

**Reply by devemux86** [reply]
Thanks everyone for your feedback! Our solutions seem to be:
- 2 column menus: with scroll in some menus (old implementation)

- 2 column menus with max 6 or 8 items to avoid scroll
and some menus will have more screens (e.g. route and Routing menu)

- The 3 column menus use smaller font

Compromises in texts are usually not nice, as they should be understandable to users.
Also Cruiser currently has 11 languages, so there are cases in all of them.

**Reply by bios4** [reply]
I'd go for option two ^^: "2 column menus with max 6 or 8 items to avoid scroll"

For the route menu I think "Roadbook" could be switched to the second screen.
And maybe there is any possibility to combine the "Section profile" and "Profile options" functions to be handled with one tile!???

For the routing menu I can imagine having the "Share" and "BRouter" tile on a second screen.

And/or: provide the "More" options ("screen switch") not with a big tile, but with smaller arrows below (I think this was the implementation in an earlier version??)

**Reply by WolfgangZ42** [reply]
> The 3 column menus use smaller font

**My preference.** 

What does Android do? It cuts the text: 
![2024-04-22 10_03_55-2024-04-22 09 03 30 png - IrfanView (Zoom_ 502 x 1004)](https://github.com/devemux86/cruiser/assets/121806238/01148327-9a80-4846-9f39-bb22dcddecdf)

Another option could be to hide the text totally - as I do in my home screen:
![2024-04-22 10_04_34-2024-04-22 09 03 48 png - IrfanView (Zoom_ 502 x 1004)](https://github.com/devemux86/cruiser/assets/121806238/86f1f64e-c474-497a-b9b1-d995614c6535)

**Reply by devemux86** [reply]
> I'd go for option two ^^: "2 column menus with max 6 or 8 items to avoid scroll"

This is also my suggestion. It could work nicely for the route menu (long press route line).

But the main Routing menu will be strange to have 2 screens.
Perhaps we can remove some items, if they are in route menu?

**Reply by devemux86** [reply]
@WolfgangZ42 making the texts 1 line is easy (if we want it).

About removing the texts, it would be difficult for new users to understand the menus.

**Reply by bios4** [reply]
> But the main Routing menu will be strange to have 2 screens.
Perhaps we can remove some items, if they are in route menu?

Maybe move the "Import", "Export" and "Share" tiles/options to the route menu?

**_EDIT:_** in the end, only the "Options", "Route diagram" and "BRouter" tiles are required in the routing menu - all others could easily belong to the route menu only!??

> About removing the texts, it would be difficult for new users to understand the menus.

Not only for new users I think. 😀

**Reply by DIrgi** [reply]
Even myself I would struggle if I only see symbols.

**Reply by DIrgi** [reply]
@devemux86 : Instead of a "more..." icon to display menu pages larger then 6 tiles, does it make sense and is it technically possible to use tabs to switch between the pages?

**Reply by devemux86** [reply]
> does it make sense and is it technically possible to use tabs to switch between the pages?

It is possible, but more complicated.
We will also lose vertical space from the tabs, that is useful especially in landscape.

**Reply by devemux86** [reply]
Another improvement is to remove the "More" buttons, gaining an extra position.

Instead add a right arrow at bottom (always visible) toolbar.
So current left arrow goes to left screen and new right arrow goes to right screen.

Remember the menu screens also have gestures:
- Swipe up / down to close them
- Swipe left to go to right screen
- Swipe right to go to left screen

**Reply by bios4** [reply]
> Remember the menu screens also have gestures

In B10 and the route menu (three-tile-view)

- swiping left CLOSES the menu when on the FIRST screen
- swiping left goes BACK to screen one when on the SECOND screen
- swiping left goes BACK to screen two when on the THIRD screen
- swiping right CLOSES the menu when on the FIRST screen
- swiping right goes BACK to screen one when on the SECOND screen
- swiping right goes BACK to screen two when on the THIRD screen

Going from one screen to the next is reliably only possible per the "More" tile, going back with the swiping right or the left-arrow at the bottom...

The very same applies to the long-press on map menu (three-tile-view)!

In the "Routing" as well as the "GPS recording" menu swiping left AND right leads back to the main menu.

**Reply by devemux86** [reply]
These will be fixed in the next update.

**Comment by devemux86** [comment]
> 2 column menus with max 6 or 8 items to avoid scroll and some menus will have more screens

> Another improvement is to remove the "More" buttons, gaining an extra position.
Instead add a right arrow at bottom (always visible) toolbar.

> Remember the menu screens also have gestures

The discussed improvements are available for testing in the latest:
- https://github.com/devemux86/cruiser/discussions/482

The gestures on the menus are described [above](https://github.com/devemux86/cruiser/discussions/496#discussioncomment-9188472).

**Reply by devemux86** [reply]
The route menu is split in 3 screens.

I don't like it very much, but it has 17 items, it doesn't fit in 2 screens, unless we reduce the items.
Some of them are also in Routing menu (now 2 screens).

**Reply by bios4** [reply]
Thanks for the new beta, swiping works again now properly.

> The route menu is split in 3 screens.
> I don't like it very much, ...

Haha - I'm also not 100% satisfied with the current order and arrangement of the tiles in the screens of the route menu.
But also with shuffling around a lot, I don't have a better idea than what you currently implemented... 😉

> unless we reduce the items

What about to have "Roadbook" and "Turn instructions" only in the routing menu?

I also would like to suggest to have "Roadbook" and "Turn instructions" together in one line in the routing menu, as well as "Import" and "Export" close together.

When no route is loaded, the routing menu currently shows (top left > top right > bottom left > bottom right):
- Options
- Import
- BRouter
- Clear

As "Clear" is not required when no route is loaded, my suggestion would be to show:
- Import
- Options
- BRouter
- Clear (greyed out)


When a route is loaded, the menu could look like:
- Waypoints
- Options
- Turn instructions
- Roadbook
- BRouter
- Clear
- Import (screen 2)
- Export (screen 2)
- Route diagram (screen 2)
- Share (screen 2)

**Reply by devemux86** [reply]
@bios4 I think we should reduce the items in the route menu to max 8+8 items and return to 2 screens.

**Reply by devemux86** [reply]
Export usually goes together with Share.
It will be strange to hide Share (more convenient to transfer content offline).

**Reply by devemux86** [reply]
If we move Export + Share in 1st routing screen, there will be 8 + 2 items in the screens.

**Reply by devemux86** [reply]
Clear can also be used to delete overlays, it is a multiple menu, not only for routes.

**Reply by devemux86** [reply]
The menu order cannot change if we have route or not, so positions are easily remembered by users.
Also import cannot be hidden in 2nd screen, should be easily found by users without too many clicks.

**Reply by bios4** [reply]
> I think we should reduce the items in the route menu

I totally agree - but what should be deleted / moved from it?
There are 28 tiles currently...

Would it be possible to rework the (currently) 5th screen (titled "Share") with its seven tiles as a modal menu which opens up after clicking on a "Share" tile?

Then we would have to deal with "only" 21 remaining menu items to rearrange... 😜

**Reply by devemux86** [reply]
>  There are 28 tiles currently...

Above the 16 or 17 route items, there are the common extra menus:
- Show address / websites
- Share in various formats

> Would it be possible to rework the (currently) 5th screen (titled "Share") with its seven tiles as a modal menu which opens up after clicking on a "Share" tile?

We could try this. Then the 3rd screen would have 5 items:
- Show address
- Show on ... x 3
- Share

**Comment by devemux86** [comment]
For "Routing" menu we could use:
Waypoins|Road book
:-:|:-:
Import|Export
Share|Options
BRouter|Clear

and as turn instructions is the less usable (?) list:
Turn instructions|Route diagram
-|-

**Comment by devemux86** [comment]
For route menu we could use:
Set via point|Set shaping point
:-:|:-:
Waypoints|Road book
Section profile|Options
Recalculate section|Recalculate route

and:
Reverse route|Place waypoints on street
:-:|:-:
Show whole route|Calculation mode
Fill with shaping points|Add favorite
Save route|Delete route

Or replace calculation mode (it is also in settings) with "Turn instructions".

**Reply by bios4** [reply]
Remove "Calculation mode" (and keep it only in settings) sounds good!

What about the following menu order for the second screen then:

| Reverse route  | Show whole route |
| :---: | :---: |
| Fill with shaping points | Place waypoints on street |
| Turn instructions | Add favorite |
| Save route | Delete route |

But still: how to handle the current 4th screen "Show address / on OSM / Google Maps / Mapillary" and the seven items in the 5th screen "Share"!?

**Reply by devemux86** [reply]
We could also do this.

Also the "Show whole route" is a bit strange to be in the route menu?
Route menu appears when we long press the route, so the route is already visible.
If we want to see more of the route, just zoom out?

The "Show whole route" already exists when we long press anywhere on the map,
so the route could be not in screen and we my want to find it.

> But still: how to handle the current 4th screen "Show address / on OSM / Google Maps / Mapillary" and the seven items in the 5th screen "Share"!?

There can be a 5th item "Share" again in the screen with the 4 "Show ..." items.
This "Share" item will not open a new screen (like now), but a quick selection dialog, so we will have 1 less screen.

(as you described [above](https://github.com/devemux86/cruiser/discussions/496#discussioncomment-9190996))

**Reply by bios4** [reply]
> "Show whole route" is a bit strange to be in the route menu?

To be honest I use this function quite frequently.
E.g. when I'm zoomed in very far to fine-set shaping points or to check a detailed routing in a densly packed area this is quite handy...

But as far as it also is available in the long-press on map menu, it could be removed here if really neccessary.

> We could try this. Then the 3rd screen would have 5 items:
> - Show address
> - Show on ... x 3
> - Share

> This "Share" item will not open a new screen (like now), but a quick selection dialog, so we will have 1 less screen.

This sounds not too bad to me.
The 3rd screen doesn't even need a certain name in its header, but could show the very same details as the first two screens...

**Reply by bios4** [reply]
What is also not optimal in addition is the "moving closing X" caused by the arrows for switching screens.
Muscle memory is a tough cookie when it comes to "intuitively pressing screen areas".

Maybe the "switching screen" arrows could be added in the center of the bottom part of the menus?

From:
<img src="https://github.com/devemux86/cruiser/assets/145460147/e6bae788-ee11-4e6f-9db5-114aff53435a" width="100%">

To:
<img src="https://github.com/devemux86/cruiser/assets/145460147/052761ce-76b1-4d45-a47e-5da66ffa300e" width="100%">

**Reply by devemux86** [reply]
This is not possible, the locations are fixed. Also they should have enough space between them.

The alternative is to return to the extra "More" buttons, but then we have to remove 1 menu item.

**Reply by WolfgangZ42** [reply]
Maybe the easy way out could be to set the X on the same position in all panels: 
![vorher](https://github.com/devemux86/cruiser/assets/121806238/369002ea-17e7-45b3-8a63-ac39f5000cb2)
For all with a hard muscle memory. 🥴

**Reply by devemux86** [reply]
Maybe we should return to the More button to avoid such complications.
Also avoid too many and duplicate menus, then all should fit in 2 screens.

I will prepare a new update.

**Reply by DIrgi** [reply]
The glove usage is for sure an argument which has to be always in consideration. Finally I assume the contact area is even bigger and not just as large as the small <-  -> signs.
For me even a swiping solution (only) to got to the next available symbols in vertical and/or horizontal direction will be fine. To show the user that there are more icon available then just the on they see, it can be shown as I.e. two small dots on the bottom (if there are two menue screens), where the dot which is in use has another color.
The menu screen itself can be closed by tapping and holding on it elsewhere for maybe one second or in any area which is out of the menue screen.

**Reply by bios4** [reply]
It seems you missed the information about swiping!??
--> https://github.com/devemux86/cruiser/discussions/496#discussioncomment-9188472

**Comment by DolomitesTom** [comment]
The new context sub sub sub menu is also confusing to me. I found the scrollable menu better. However, it is better to scroll in full tiles and not in half.

**Comment by devemux86** [comment]
The latest Beta has more simple menus without too many or duplicate items.

I wonder do we need all the "Show" and "Share" items in the more menu?

**Reply by bios4** [reply]
The minimized route menu looks nice @devemux86 👍... 

With one exception: I really would appreciate to have the "Waypoints' tile back also in the route menu to organize/(re-)order them accordingly directly from the map and without the detour via Main > Routing.

You could
- either use the one free "slot" in the second screen for it (my preference)
- or add it instead of "Show whole route" (which is also available in the long-press on map menu)

This would be really great!

About the sharing options: I did not use any of these entries ever, except "Show on Google Maps". I don't know if other users need these more often...

**Reply by bios4** [reply]
Swiping "back" (right) from the Share-menu to screen 2 of the route menu is not working, this is only possible with the back-arrow...

**Reply by devemux86** [reply]
Share screen is a native Android screen, it does not support gestures.
(like the custom screens with the buttons)

**Reply by bios4** [reply]
> have the "Waypoints' tile back also in the route menu

Suggested change of the second screen of the route menu:

| Waypoints | Reverse route |
| :---: | :---: |
| Place waypoints on street  | Fill with shaping points |
| Show whole route | Add favorite |
| Save route | More... |

And: do we really need the "Add favorite" tile in the route menu?

The feature is already available in long-press on map menu, from there you can even freely set a cursor where you want the favorite to be.

Whereas when choosing it from the route menu, the very position where I tapped on the map is predefined (without the option of placing the cursor to select the detailed position).

At least the option to place a cursor should also be available when selecting to add a favorite from the route menu...

**Comment by DolomitesTom** [comment]
Hmm, I am also missing the "Waypoints' Tile. The last menu page is somehow a break in style. Perhaps you could show these as buttons, but then eight on one page and not two times four. I also didn't realise that swiping left and right works to change the menu. That's great. Then there is no need for a more button, the arrow at the bottom left and right shows that there is another menu. And then there is space for the waypoints tile and additionally the import and export tile.

On the first Menu page there is missing the arrow to the right.

**Comment by devemux86** [comment]
We could add the "Waypoints" again, as it is really an edit tool for the route.
Also indeed the "Add favorite" can be removed, as it is available elsewhere.

About the "Routing" menu, should we move "Waypoints" + "Road book" row at the top?

**Reply by bios4** [reply]
> We could add the "Waypoints" again, as it is really an edit tool for the route.

👍 ❤️ 

> About the "Routing" menu, should we move "Waypoints" + "Road book" row at the top?

What about this in Main > Routing:
| Waypoints | Road book |
| :---: | :---: |
| Turn instructions | BRouter |
| Import | Export |
| Share | Clear |

**Reply by devemux86** [reply]
@bios4 thanks, note when not using BRouter, this will move Import at 2nd row.

Although most should prefer the BRouter offline routing and options! :slightly_smiling_face:

**Reply by bios4** [reply]
Could you make the tile "greyed out"/inactive in that case?
Like you already do with the different "Recalculate"-tiles in the route menu based on set recalc. option?

If not, we should move the "Share"-tile to that position, so that "Import" and "Export" stay at the same position in all cases...

**Reply by devemux86** [reply]
Or could be:

Waypoints|Road book
:-:|:-:
Import|Export
Share|Turn instructions
BRouter|Clear

**Reply by bios4** [reply]
I'm fine with either option. 😀

I just wonder what is the "switch" to either "deactivate" a tile (but still show it greyed out) or hide it at all.
This is somehow inconsistent (and not neccessarily self-explaining/intuitive).

**Reply by devemux86** [reply]
There are also actions that only work on active route, so alternatives have fewer actions.
It is often a simpler menu if it only contains the available actions. If we "disable" buttons it can get confusing.

Like here, if someone wants to use OpenRouteService or Valhalla,
why they should also see the BRouter or GraphHopper tiles?

**Reply by bios4** [reply]
> It is often a simpler menu if it only contains the available actions. If we "disable" buttons it can get confusing.

I fully agree... 👍

**Reply by devemux86** [reply]
Similarly the map menu does not show the offline map tiles, unless an offline map is used.
Although someone may suggest to show all tiles (even disabled), so users can see the offline map benefits!
(options for theme, layers, language, etc)

**Reply by bios4** [reply]
Nice update in B13! 👍

> The last menu page is somehow a break in style.

As there are only eight show/share entries left, you could really think about making it again a "default" 8-tile menu...
Because there is no "return back" tile required but only the back arrow below we should be fine with the eight show/share tiles on a 3rd screen!?

Then swiping could be also made possible again for the show/share entries.

**Reply by devemux86** [reply]
I'm still thinking if we really need all these options especially on Android.

**Comment by devemux86** [comment]
I wonder if we need the "Export" and "Share" in the main routing menu
or we could move them to the 2nd screen of route menu next to "Save"?

(as they are not that often used actions)

**Reply by devemux86** [reply]
Any thoughts on this?

**Reply by DolomitesTom** [reply]
Yes in the 2nd screen would be fine

**Reply by devemux86** [reply]
> I wonder if we need the "Export" and "Share" in the main routing menu
or we could move them to the 2nd screen of route menu next to "Save"?

Let's try it in the next update.

**Reply by WolfgangZ42** [reply]
In my opinion, where there is "Import" there should also be "Export" - otherwise it leads to confusion. 
I think "Share" could be moved to the **2nd route menu**. 
The name of the **3rd route menu** "Share" can also lead to confusion. Admittedly, you can only get there by swiping and it is not identical to the "Share" function from the **main routing menu** (or later **2nd route menu**). 
What strikes me is the redundancy of "Waypoints" in the **main routing menu** and in the **2nd route menu**, where it is more logically located in my opinion.

**Reply by devemux86** [reply]
Export and Share are similar with different target, so they cannot really be separate.
It would be nice if Android allowed us to share in file system and remove the Export.

Waypoints: also road book and turn instructions and all menus that exist only when we have a route.
We cannot put everything in the route menu, unless there is free space with no more than 2 screens.

Anyway the thought is that actions that are not used often should not occupy space in first screens.
Probably we will see how well any idea works in action. :slightly_smiling_face:

**Reply by devemux86** [reply]
> What strikes me is the redundancy of "Waypoints" in the main routing menu and in the 2nd route menu, where it is more logically located in my opinion.

Instead, if we have a new route tool in the future, waypoints could be removed as duplicate from the route menu. :slightly_smiling_face:

**Reply by bios4** [reply]
> The name of the 3rd route menu "Share" can also lead to confusion. 

Do you mean the "Share" function from the l**ong-press on map menu**?
Because in the route menu there is (currently) no share option...

But I agree that it is a bit confusing that both tiles (having the same name and icon) offer two different functions, depending on where they are called from.

**Reply by devemux86** [reply]
Share has a standard icon and text.
What is extra in the 2nd case is the show in external apps, which on mobile does not (?) seem very useful.

**Reply by WolfgangZ42** [reply]
> The name of the 3rd route menu "Share" can also lead to confusion. **_Admittedly, you can only get there by swiping_** 

Maybe this is unintentionally left over from a beta version?

**Reply by devemux86** [reply]
> Maybe this is unintentionally left over from a beta version?

Correctly, this was fixed in the latest update.

**Reply by devemux86** [reply]
Please check in the latest Beta if you like this or the previous implementation.

**Reply by bios4** [reply]
> which on mobile does not (?) seem very useful.

At least "copy coordinates" and "show on Google Maps" are features I would use regularly.
I probably won't use any of the others ever.

How frequently do other users need these options?!?

**Reply by bios4** [reply]
> Please check in the latest Beta if you like this or the previous implementation.

On first sight it might confuse users that both tiles ("Share" and "Export") in the route menu open an identical looking screen in the first step and then offer different features.

Apart from that the move of the tiles from the routing menu to the route menu makes sense.

**Comment by DolomitesTom** [comment]
Where do I find now the show location on Google maps and share as plus code?
Found it, forget the question

**Comment by devemux86** [comment]
So do you like the "Export" and "Share" in the route menu?

Or do you prefer their previous position in the "Routing" menu?

**Reply by WolfgangZ42** [reply]
I would prefer Export in the Routing Menu - also Share if you wont split the two. See https://github.com/devemux86/cruiser/discussions/496#discussioncomment-9263180
You ask for opinions - this is mine.

**Reply by bios4** [reply]
Let's see if there will be enough answers to decide what's best - because as already mentioned [here](https://github.com/devemux86/cruiser/discussions/496#discussioncomment-9263955) my opinion is:

> Apart from that the move of the tiles from the routing menu to the route menu makes sense.

**Comment by DolomitesTom** [comment]
I also would appreciate the 2 functions **additionally** in the route menu. It is not a problem to have redundant functions in different places, it's a benefit in my point of view. Instead of reducing menus why not having same important functions double in different locations?

**Reply by bios4** [reply]
Counter question (serious): why have it doubled? What's the benefit?

**Reply by devemux86** [reply]
Duplicate actions in many places may seem convenient at first,
but in the end they complicate the interface by loading it with extra menus and users get confused.

Also, as the available places are not unlimited and new tools are planned in the future, we have to be careful. :slightly_smiling_face:

**Comment by devemux86** [comment]
A simpler solution is to have 1 menu for Export / Share with sub-menu for the specific action.
Then we can return it in the Routing menu. Also GPS recording menu will have one less item.

Let's try it in the next update.

**Comment by 0709wiwiwi** [comment]
I don't understand why the menu reorganization is being discussed **so persistently**.
When new features are introduced, **all of this** will be turned upside down again anyway.

**Comment by devemux86** [comment]
<img src="https://github.com/devemux86/cruiser/assets/3484020/c1b5e886-2124-420d-acd2-e5628d04ccd8" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/e301f583-376e-44b7-9a36-4e025ff27a28" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/88c5a269-469a-4d89-8463-3919be938866" width="33%"/>

**Reply by bios4** [reply]
Which screenshot represents which (new) menu ^^ ❓

1st - Main > Routing?
2nd?
3rd - screen after clicking "Export / Share" in any other menu?

**Reply by devemux86** [reply]
1. Routing menu
2. GPS recording menu
3. After click "Export / Share" menu

**Reply by bios4** [reply]
> 2. GPS recording menu

Thanks.
I was confused by the "Track diagram" tile, but just recognized that this is not the "show diagram yes/no" tile but the other one...

How does the GPS recording menu look like when a recording is currently active/running?

And how does it look like when there is no recorded GPS track currently available? Are the "Export / Share", "Save track" and "Clear" tiles also available then or are they hidden?

**Reply by devemux86** [reply]
GPS recording - on|GPS recording - empty
-|-
![1](https://github.com/devemux86/cruiser/assets/3484020/e4839021-f248-4044-ac00-f23b40fad412)|![2](https://github.com/devemux86/cruiser/assets/3484020/7abb3d42-95b4-4e52-ac3c-6692d8b6108a)

**Comment by devemux86** [comment]
Everyone please avoid posting new questions in edits.
The edits do not generate new notifications or emails, so the questions may be lost.

Better write them in new posts.

**Comment by devemux86** [comment, marked as answer]
Menu improvements are available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
